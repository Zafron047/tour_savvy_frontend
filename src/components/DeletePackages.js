import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { deletePackage, fetchPackages } from '../redux/packages/packagesSlice';
import { getReservations } from '../redux/reservations/reservationSlice';
import DeletePackageComp from './DeletePackageComp';

const DeletePackages = () => {
  const packages = useSelector((state) => state.packages.allPackages);
  const loading = useSelector((state) => state.packages.loading);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchPackages());
  }, [dispatch]);
  const handleDelete = (packageId) => {
    dispatch(deletePackage(packageId)).then(() => {
      dispatch(getReservations());
    });
  };
  const user = JSON.parse(localStorage.getItem('user'));
  if (!user) {
    return <div>Please log in to delete a package.</div>;
  }
  if (loading) {
    return (
      <>
        <div>
          <h2>Loading...</h2>
        </div>
      </>
    );
  }
  return (
    <div className="container delete-wrap">
      {packages.map((p) => (
        <div className="card-container" key={p.id}>
          <div className="per-package" key={p.id}>
            <DeletePackageComp key={p.id} p={p} />
            <button type="button" onClick={() => handleDelete(p.id)} className="btn btn-danger">Delete Package</button>
          </div>
          <div className="description" key={p.id}>
            {p.description}
          </div>
        </div>
      ))}
    </div>
  );
};
export default DeletePackages;
