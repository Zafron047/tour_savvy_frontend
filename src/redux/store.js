import { configureStore } from '@reduxjs/toolkit';
import packagesSlice from './packages/packagesSlice';
import reservationsReducer from './reservations/reservationSlice';
import userSlice from './authentication/userSlice';

const loadStateFromLocalStorage = () => {
  try {
    const serializedState = localStorage.getItem('reduxState');
    if (serializedState === null) {
      return undefined;
    }
    return JSON.parse(serializedState);
  } catch (err) {
    return undefined;
  }
};

const loadUserFromLocalStorage = () => {
  try {
    const serializedUser = localStorage.getItem('user');
    if (serializedUser === null) {
      return undefined;
    }
    return JSON.parse(serializedUser);
  } catch (err) {
    return undefined;
  }
};

const store = configureStore({
  reducer: {
    packages: packagesSlice,
    reservations: reservationsReducer,
    current_user: userSlice,
  },
  preloadedState: {
    current_user: loadUserFromLocalStorage(),
    ...loadStateFromLocalStorage(),
  },
  reservations: reservationsReducer,
});

export default store;
