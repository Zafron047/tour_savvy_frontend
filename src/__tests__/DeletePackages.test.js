import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import DeletePackages from '../components/DeletePackages';

test('Should render DeletePackages component correctly', () => {
  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <DeletePackages />
      </BrowserRouter>
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
