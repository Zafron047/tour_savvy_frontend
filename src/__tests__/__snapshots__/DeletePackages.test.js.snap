// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render DeletePackages component correctly 1`] = `
<div>
  <div>
    Please log in to delete a package.
  </div>
</div>
`;
